src/HttpMessage.php

```diff
@@ -208,7 +208,7 @@ public function hasHeader(string $name): bool
 
     private function isNameValid(string $name): bool
     {
-        return (bool) \preg_match('/^[A-Za-z0-9`~!#$%^&_|\'\-:]+$/', $name);
+        return (bool) \preg_match('/^[A-Za-z0-9`~!#$%^&_|\'\-:.]+$/', $name);
```

Comment:
Seems like a typo, because `:` shouldn't be in this list.

Reply:
Additional characters missing: `*` and `+`, see relevant spec: https://www.rfc-editor.org/rfc/rfc7230#appendix-B

Reply:
I added only dot `.`. See diff.

Reply:
I know, but we should also add the other two and remove `:`